core/src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -50,7 +50,7 @@
     /**
      * Sets the arbitrary name to be assigned to the query (see named queries).
      */
-    QB queryName(String queryName);
+    QueryBuilder queryName(String queryName);
```

Comment:
This is why it had the generics. OTOH I'll bet if we always return the subclass of QueryBuilder anyplace interesting and have each of them implement this method by returning their own type then no client code will have to change. We should probably add something to the docs on this method telling implementers to return their own type, not QueryBuilder.